Makefile

```diff
@@ -29,13 +44,19 @@ BAZEL_STARTUP_ARGS ?=
 BAZEL_BUILD_ARGS ?=
 BAZEL_TEST_ARGS ?=
 
+# Environment for tests, the directory containing istio and deps binaries.
+# Typically same as GOPATH/bin, so tests work seemlessly with IDEs.
+export ISTIO_BIN=${GOPATH}/bin
+
 hub = ""
 tag = ""
 
 ifneq ($(strip $(HUB)),)
 	hub =-hub ${HUB}
 endif
 
+# If tag not explicitly set in users' .istiorc or command line, default to the git sha.
+TAG ?= $(shell git rev-parse --verify HEAD)
```

Comment:
This won't work when users download a tarball of a released version. Should the fallback be the version number (like having a VERSION file)?

Reply:
That's what the presubmit is doing - we can change it later, but as is TAG can be set by user either 
via command line or env variable - this is only the default.